File: heart/src/main/java/com/vantagecircle/heartrate/data/DataManager.java
package com.vantagecircle.heartrate.data;

import android.content.Context;

import com.vantagecircle.heartrate.model.History;
import com.vantagecircle.heartrate.annotation.ApplicationContext;

import java.util.ArrayList;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by SiD on 11/11/2017.
 */
@Singleton
public class DataManager {
    private Context mContext;
    private DatabaseHelper mDatabaseHelper;

    @Inject
    public DataManager(@ApplicationContext Context mContext, DatabaseHelper mDatabaseHelper) {
        this.mContext = mContext;
        this.mDatabaseHelper = mDatabaseHelper;
    }

    public boolean insertHistory(History history) {
        return mDatabaseHelper.insertHistory(history);
    }

    public ArrayList<History> getHistory() {
        return mDatabaseHelper.getHistory();
    }

    public ArrayList<History> getHistory(int limit) {
        return mDatabaseHelper.getHistory(limit);
    }
}

File: heart/src/main/java/com/bapidas/heartrate/data/DataManager.kt
package com.bapidas.heartrate.data

import com.bapidas.heartrate.data.model.HistoryModel
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by SiD on 11/11/2017.
 */
@Singleton
class DataManager @Inject constructor(
    private val mDatabaseHelper: DatabaseHelper
) {
    fun insertHistory(historyModel: HistoryModel): Boolean {
        return mDatabaseHelper.insertHistory(historyModel)
    }

    val history: ArrayList<HistoryModel>
        get() = mDatabaseHelper.history

    fun getHistory(limit: Int): ArrayList<HistoryModel> {
        return mDatabaseHelper.getHistory(limit)
    }
}